### PR TITLE
*: use std slices to replace exp/slices

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: ref #45933

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by lance6716 — APPROVED** [review]

File: planner/core/tiflash_selection_late_materialization.go [hunk 1]
```diff
@@ -144,8 +146,15 @@ func groupByColumnsSortBySelectivity(sctx sessionctx.Context, conds []expression
 	}
 
 	// Sort exprGroups by selectivity in ascending order
-	slices.SortStableFunc(exprGroups, func(x, y expressionGroup) bool {
-		return x.selectivity < y.selectivity || (x.selectivity == y.selectivity && len(x.exprs) < len(y.exprs))
+	slices.SortStableFunc(exprGroups, func(x, y expressionGroup) int {
+		c := cmp.Compare(x.selectivity, y.selectivity)
+		if c < 0 {
+			return c
+		}
```
Comment (Defined2014) [inline planner/core/tiflash_selection_late_materialization.go:153]: Those three lines are useless?

Comment (hawkingrei) [reply planner/core/tiflash_selection_late_materialization.go]: I refactored it.

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**